### PR TITLE
[ci] use service accounts instead of workload identity federation

### PR DESCRIPTION
**Pull request by nbdd0121**
It appears that Bazel cache performance suffers when using WIF. This is likely a bug in Bazel (or google-auth-library java) since both the ID token and the exchanged access token can be cached. I suppose when WIF is in use there're excessive amount of access token exchange happening which slows down all Google bucket accessese.

As a workaround, go back to use service account JSON instead of WIF.

### CODE REVIEW

File: .github/actions/prepare-env/action.yml [hunk 1]
```diff
@@ -6,6 +6,9 @@ name: Prepare environment
 description: Install dependencies and prepare environment needed for OpenTitan
 
 inputs:
+  service_account_json:
+    description: Service account JSON for Google Cloud access
+    default: ''
```
Comment (jwnrt) [inline .github/actions/prepare-env/action.yml:11]: Can this default to `'${{ secrets.BAZEL_CACHE_CREDS }}'`? We're only ever going to pass this value from callers

Comment (nbdd0121) [reply .github/actions/prepare-env/action.yml:11]: Actions can't have access to secrets directly, only workflows can.

Comment (jwnrt) [reply .github/actions/prepare-env/action.yml:11]: I guess that makes sense

**Review by jwnrt — APPROVED** [review]

**Comment by nbdd0121** [comment]
Merge to fix sw build & test CI, skipping bitstream build CI.

**Pull request merged**